ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java

```diff
@@ -282,6 +282,7 @@ private CompactionType checkForCompaction(final CompactionInfo ci,
     }
 
     if (runJobAsSelf(runAs)) {
+      ci.runAs = runAs;
```

Comment:
Why do we need this?

Reply:
without setting explicitly here, insert query find it null and skip the [here](https://github.com/rajkrrsingh/hive/blob/3d12959339b22becee0aa986852049b46867f016/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java#L5288)